core/types/transaction_signing.go

```diff
@@ -260,7 +260,7 @@ func (s eip2930Signer) Sender(tx *Transaction) (common.Address, error) {
 		if !tx.Protected() {
 			return HomesteadSigner{}.Sender(tx)
 		}
-		V = new(big.Int).Sub(V, s.chainIdMul)
+		V = new(big.Int).Sub(V, new(big.Int).Mul(tx.ChainId(), big.NewInt(2)))
```

Comment:
I get that for external transactions, we technically want a "replay", if the transaction is from another region or zone in our network. But, wouldn't this also allow replays from other networks? e.g. a transaction from testnet would pass this check on mainnet, no?

Reply:
we have different chainIds for mainnet and testnet right

Reply:
The problem here, is that a transaction from a different network (e.g. mainnet vs testnet) would still be valid on the other network. I think we need to disambiguate between chainId and networkId here, since we want to allow signatures from many chains, but only the chains in _our_ network.

_our_, meaning the verifier's context

Reply:
We bucket the chainIDs by Network here
![Uploading Screen Shot 2022-04-27 at 12.10.13 AM.png…]()
k

Reply:
I'm unable to view your screenshot. Could you repost?

Reply:
@shreekarashastry add a check to make sure the the tx.ChainId() is in the chainIDRange

Reply:
It's also worth mentioning, that performing an extra field multiplication here will have a potentially significant performance impact, as this operation may be invoked thousands of times per block. It might be preferred to create an array of precomputed `chainIdMul` values for each of our allowed chains (as they do for `s.chainIdMul`, instead of computing the new `chainIdMul` JIT like this.

This is just a performance optimization though, so I wouldn't consider it necessary.

Reply:
@alanorwick @wizeguyy Here is my proposal, We update the EIP 155signer to store the chainIdMul for all 13 different chainID mul's precalculated.  So making it an array. This will reduce the number of field multiplications significantly.

Reply:
Sounds good to me too!